compiler+runtime/src/cpp/jank/runtime/obj/persistent_string_sequence.cpp

```diff
@@ -0,0 +1,93 @@
+#include <jank/runtime/obj/persistent_string_sequence.hpp>
+
+namespace jank::runtime
+{
+  obj::persistent_string_sequence::static_object(obj::persistent_string_ptr const s)
+    : str{ s }
+  {
+    assert(!s->data.empty());
+  }
+
+  obj::persistent_string_sequence::static_object(obj::persistent_string_ptr const s, size_t const i)
+    : str{ s }
+    , index{ i }
+  {
+    assert(!s->data.empty() && i < s->data.size());
+  }
+
+  /* behavior::objectable */
+  native_bool obj::persistent_string_sequence::equal(object const &o) const
+  {
+    return detail::equal(o, str->data.begin(), str->data.end());
```

Comment:
Bug here: missing `+ index`. This will mean that `(= (next (seq "foo")) (seq "foo"))`, which is incorrect.